lib/activerecord-import/import.rb

```diff
@@ -205,9 +205,10 @@ def import( *args )
       # Force the primary key col into the insert if it's not
       # on the list and we are using a sequence and stuff a nil
       # value for it into each row so the sequencer will fire later
-      if !column_names.include?(primary_key) && connection.prefetch_primary_key? && sequence_name
-         column_names << primary_key
-         array_of_attributes.each { |a| a << nil }
+      if ![primary_key].flatten.*.to_sym.to_set.subset?(column_names.*.to_sym.to_set) &&
```

Comment:
Does that line of code work? I think something like this is a little easier to grok:

```
symbolized_column_names = Array(column_names).map(&:to_sym)
symbolized_primary_key = Array(primary_key).map(&:to_sym)

if !symbolized_primary_key.to_set.subset?(symbolized_column_names.to_set) && ..
```